liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+    const oracleABI = getAbi(config.liquidationMonitor.oracleABI);
+    const oneInchABI = getAbi(config.liquidationMonitor.oneInchABI);
+    data.comptrollerContract = new ethers.Contract(
+      comptrollerAddress,
+      comptrollerABI,
+      data.provider,
+    );
+    data.oneInchContract = new ethers.Contract(
+      oneInchAddress,
+      oneInchABI,
+      data.provider,
+    );
+    // Unused but available for the future
+    data.oracleContract = new ethers.Contract(
+      oracleAddress,
+      oracleABI,
+      data.provider,
+    );
+
+    /* eslint-enable no-param-reassign */
+    // #endregion
+
+    // #region Get initial accounts from Compound API
+    // To-do: Can be bootstrapped with ~30 eth_getLogs calls,  500k blocks per call, using filter:
+    // Borrow (address borrower, uint256 borrowAmount, uint256 accountBorrows, uint256 totalBorrows)
+    // topic0: 0x13ed6866d4e1ee6da46f845c46d7e54120883d75c5ea9a2dacc1c4ca8984ab80
+
+    // Simple bootstrap with Compound for now
+    const apiURL = 'https://api.compound.finance/api/v2/account';
+
+    // Helper for Compound API
+    function buildJsonRequest(maxHealth, minBorrow, pageNumber, pageSize) {
+      const jsonRequest = {
+        addresses: [], // returns all accounts if empty or not included
+        block_number: 0, // returns latest if given 0
+        max_health: { value: maxHealth },
+        min_borrow_value_in_eth: { value: minBorrow },
+        page_number: pageNumber,
+        page_size: pageSize,
+      };
+      return jsonRequest;
+    }
+
+    // Find total number of results with the first request
+    const initialRequest = buildJsonRequest(
+      maximumHealth,
+      minimumBorrowInETH,
+      1,
+      1,
+    );
+    const initialResults = await callAPI(apiURL, initialRequest);
+    const totalEntries = initialResults.pagination_summary.total_entries;
+
+    // Determine number of pages needed to query. Results vs config limit.
+    const maxEntries = Math.min(maxTrackedAccounts, totalEntries);
+    const maxPages = Math.ceil(maxEntries / 100);
+
+    // Query each page and add accounts. Starting at 1 and including maxPages
+    const accounts = [];
+    const pages = [...Array(maxPages)].map((_, i) => 1 + i);
+    await Promise.all(pages.map(async (page) => {
+      const currentRequest = buildJsonRequest(
+        maximumHealth,
+        minimumBorrowInETH,
+        page,
+        100, // 100 results per page was optimal
+      );
+      const apiResults = await callAPI(apiURL, currentRequest);
+      apiResults.accounts.forEach((account) => {
+        accounts.push(account);
+      });
+    }));
+    // #endregion
+
+    // #region Parse Compound data into new objects
+    // Get a unique list of token addresses
+    ts('Processing tokens now');
+    const tokenSet = new Set();
+    accounts.forEach((account) => {
+      account.tokens.forEach(async (token) => {
+        tokenSet.add(token.address);
+      });
+    });
```

Comment:
I don't know enough about the implementation of `Set`, but have you considered constructing the Array and then passing it to the `Set` constructor instead?  I'm guessing that any performance benefit will rely upon how many duplicates you expect (the number of accounts and tokens is probably so low as to be in the noise of any performance difference).

@ae-cloin may have some insight, but this is my first crack at it:
```suggestion
    const tokens = accounts.map((account) => account.tokens.map((token) => token.address)).flat();
    const tokenSet = new Set(tokens);
```